app/assets/javascripts/analytics.js

```diff
@@ -0,0 +1,209 @@
+      .attr("height", this.height + this.margins.bottom);
+
+    this.setYScale(this.data);
+    this.draw();
+
+    if(this.labels) {
+      this.drawLabels();
+    }
+
+    if(this.xAxis) {
+      this.drawAxis();
+    }
+  }
+
+  update () {
+    this.setYScale(this.data);
+
+    this.svg.selectAll(".bar")
+      .data(this.data)
+      .transition()
+      .duration(750)
+      .attr("height", (d) => {
+        return this.scale(d.value);
+      })
+      .attr("y", (d) => {
+        return this.height - this.scale(d.value);
+      });
+
+    this.drawLabels();
+
+    this.svg.selectAll(".label")
+      .data(this.data)
+      .transition()
+      .duration(1000)
+      .text((d) => {
+        return d.value;
+      })
+      .attr("y",    this.setYForLabel.bind(this)    )
+      .attr("fill", this.setFillForLabel.bind(this) );
+  }
+
+  setYForLabel (d) {
+    let scaled  = this.scale(d.value),
+        y       = this.height - scaled + 15;
+
+    if(scaled < 20 ) {
+      y -= 20;
+    }
+
+    return y;
+  }
+
+  setFillForLabel (d) {
+    return (this.scale(d.value) < 20) ? '#333' : '#fff';
+  }
+
+  draw () {
+    this.svg.selectAll(".bar")
+      .data(this.data)
+      .enter()
+      .append("rect")
+      .attr("width", this.width / this.data.length - this.barPadding)
+      .attr("fill", this.fill)
+      .attr("class", 'bar')
+      .attr("height", (d) => {
+        return this.scale(d.value);
+      })
+      .attr("y", (d) => {
+        return this.height - this.scale(d.value);
+      })
+      .attr("x", (d, i) => {
+        return i * (this.width / this.data.length);
+      });
+  }
+
+  drawLabels () {
+    this.svg.selectAll("text")
+      .data(this.data)
+      .enter()
+      .append("text")
+      .text((d) => {
+        return d.value;
+      })
+      .attr("x", (d, i) => {
+        return i * (this.width / this.data.length) + (this.width / this.data.length - this.barPadding) / 2;
+      })
+      .attr("y", this.setYForLabel.bind(this) )
+      .attr("class", "label")
+      .attr("text-anchor", "middle")
+      .attr("font-family", "sans-serif")
+      .attr("font-size", "11px")
+      .attr("fill", this.setFillForLabel.bind(this) );
+  }
+
+  drawAxis () {
+    var xScale = d3.scale.ordinal()
+                   .domain(this.data.map( (d, i) => {
+                     return moment(d.date).format(this.axisDateFormat);
+                   }))
+                   .rangeBands([0, this.width]);
+
+
+    var xAxis = d3.svg.axis()
+                  .scale(xScale)
+                  .orient('bottom')
+
+
+    this.svg.append("g")
+        .attr("class", "x axis")
+        .attr("transform", `translate(0, ${(this.height)})`)
+        .call(xAxis);
+  }
+
+  setYScale (dataset) {
+    this.scale = d3.scale.linear()
+                   .domain([0, d3.max( dataset, (d) => {
+                     return d.value;
+                   })])
+                   .range( [0, this.height]);
+
+  }
+}
+
+class Dashboard {
```

Comment:
the relationship between `Dashboard` and `AnalyticsDashboard` is a little confusing - it would be great to maybe have a comment in this file.